crates/uv-resolver/src/pubgrub/report.rs

```diff
@@ -407,67 +407,59 @@ impl PubGrubReportFormatter<'_> {
-            let PubGrubPackageInner::Package { name, .. } = &**package else {
-                return false;
-            };
-            selector.prerelease_strategy().allows(name)
-        }
-
         let mut hints = IndexSet::default();
         match derivation_tree {
             DerivationTree::External(external) => match external {
                 External::Custom(package, set, _) | External::NoVersions(package, set) => {
-                    // Check for no versions due to pre-release options
-                    if let Some(selector) = selector {
-                        let any_prerelease = set.iter().any(|(start, end)| {
-                            let is_pre1 = match start {
-                                Bound::Included(version) => version.any_prerelease(),
-                                Bound::Excluded(version) => version.any_prerelease(),
-                                Bound::Unbounded => false,
-                            };
-                            let is_pre2 = match end {
-                                Bound::Included(version) => version.any_prerelease(),
-                                Bound::Excluded(version) => version.any_prerelease(),
-                                Bound::Unbounded => false,
-                            };
-                            is_pre1 || is_pre2
-                        });
-
-                        if any_prerelease {
-                            // A pre-release marker appeared in the version requirements.
-                            if !allowed_prerelease(package, selector) {
-                                hints.insert(PubGrubHint::PreReleaseRequested {
-                                    package: package.clone(),
-                                    range: self.simplify_set(set, package).into_owned(),
-                                });
-                            }
-                        } else if let Some(version) =
-                            self.available_versions.get(package).and_then(|versions| {
-                                versions
-                                    .iter()
-                                    .rev()
-                                    .filter(|version| version.any_prerelease())
-                                    .find(|version| set.contains(version))
-                            })
-                        {
-                            // There are pre-release versions available for the package.
-                            if !allowed_prerelease(package, selector) {
-                                hints.insert(PubGrubHint::PreReleaseAvailable {
-                                    package: package.clone(),
-                                    version: version.clone(),
-                                });
+                    if let PubGrubPackageInner::Package { name, .. } = &**package {
+                        // Check for no versions due to pre-release options
```

Comment:
The whole body is now just indented under `if let PubGrubPackageInner::Package { name, .. } = &**package`. Unfortunately we can't do this in the `match` itself because of the inner struct.